src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = findIndex(hash(key));
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> iterator = table[index];
+            while (iterator.next != null || Objects.equals(iterator.key, key)) {
```

Comment:
```suggestion
            while (iterator != null) {
```
let's try to remove that upper `if`

Reply:
Changed